README
NAME
    App::Mobirc - pluggable IRC to HTTP gateway

DESCRIPTION
    mobirc is a pluggable IRC to HTTP gateway.

AUTHOR
    Tokuhiro Matsuno <tokuhirom@gmail.com>

    and Mobirc AUTHORS.

LICENSE
    Copyright (C) Tokuhiro Matsuno

    This library is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself.



Changes
{{$NEXT}}

4.07 2013-04-10T23:24:44Z

    - move tools/ directory to author/ for pause indexing issue.
      (tokuhirom)

4.06 2013-04-09T05:53:39Z


    - Fixed packaging issue
      (Reported by kazeburo++)

4.05 2013-04-05T08:26:19Z

    - some tiny fix(mattn)

4.04

    - fixed testing issue

4.03

    - fixed testing issue

4.02

    - fixed testing issue
    - fixed module loading issue

4.01

    - fixed test cases
    - and misc bug fixes introduced on 4.00.

4.00

    - support multiple servers
    - removed android view(is not maintenable)
    - removed mobile-ajax view(is not maintenable)

3.00

    - switch to plack based.
    - switch to AnyEvent instead of POE.
    - do not use 'mobile phone id' for security reason.
    - use Router::Simple, instead of HTTPx::Dispatcher
    - rebuilt extlib/ with cpanm

1.99_02

    2.00 RC2

    - fixed f*cking utf8 flag related bug.(reported by otsune++)
    - rebirth middleware support
    - bump up version:
        Mouse                 0.16
        YAML                  0.68
        HTTP::MobileAttribute 0.13
        HTTP::Engine          0.1.2
        HTTP::Session         0.29
    - added deps:
        MouseX::Types 0.01
    - bump up jQuery version 1.2.6 => 1.3.1

1.99_01

    ready to 2.00

    - configuration file is moved to ini from YAML
    - import H::SQ::DoCoMoGUID for docomo phones
    - load HTMLFilter::DoCoMoCSS and MessageBodyFilter::IRCColor automatically

1.08_02

    - oops. miss..

1.08_01

    - use Mouse instead of Moose
    - less deps
        - don't depend to Data::Visitor::Encode
        - don't depend to XML::LibXML
        - don't depend to HTTP::MobilAgent
        - don't depend to String::TT
        - don't depend to HTML::Entities::ConvertPictograms
    - purge cp932 support
    - less memory
    - load two plugins automatically
        HTMLFilter::DoCoMoCSS
        MessageBodyFilter::IRCColor
    - show http server's banner at startup
    - use HTML::TreeBuilder::XPath instead of XML::LibXML
      (for less deps for XS)
    - use T::MicroTemplate, instead of Template::Declare
      (designer friendly, less memory, fast)
    - don't depend to Proc::Daemon
    - config.httpd.* was moved to config.global.*
    - move repo to github
    - use HTTP::MobileAttribute instead of HTTP::MobileAgent
    - purge HTMLFilter/ConvertPictograms plugin.
    - use MD5 instead of SHA1 for generate session key
    - purge Plugin::GPS

1.07

    - change authentication flow
    - use HTTP::Session
    - Plugin::Component::Twitter: deprecated
    - better iphone mode(retlet++)

1.06

    - bump up deps to PoCoIRC 5.88
      remove quick & dirty hack for
      http://rt.cpan.org/Ticket/Display.html?id=38773
      because this bug fixed at PoCoIRC 5.88
      Bingos++ # fast response

1.05

    - use HTTP::Engine
    - depended to POE::Component::IRC 5.86
    - quick & dirty bug fix for
      http://rt.cpan.org/Ticket/Display.html?id=38773

1.04

    - no Compat
    - use HTTP::Engine 0.0.14

1.03

    - rebirth recent_log_per_page(by hirose31++)
    - updated dependencies
      Data::Visitor::Encode
      Template
    - use HTTP::Engine::Compat

1.02

    - added iPhone mode(contributed by retlet++)
    - added http_link_target option for Plugin::MessageBodyFilter::Clickable
      (contributed by aerith++)
    - depend to latest version of Moose(0.48 => 0.54)

1.01

    - Plugin::HTMLFilter::DoCoMoCSS:
      bug fixed.
      this module does not working... broken in 0.99_01.
      (reported by kga++ and patch by chiba++)
   - generate valid xhtml
   - depend to HTTP::Engine 0.0.11(0.0.10 is borken =( )

1.00

    - MAJOR VERSION UP!

0.99_06

    - THIS IS DEVELOPER RELEASE!
    - irssi support

0.99_05

    - THIS IS DEVELOPER RELEASE!
    - support japanese channel
    - fixed some dependencies
    - added new plugin: ChannelPageParts

0.99_04

    - THIS IS DEVELOPER RELEASE!
    - App::Mobirc::HTTPD => App::Mobirc::Plugin::Component::HTTPD
    - bug fixed around Authorizer::BasicAuth
    - rebirth the mobile-ajax with T::D

0.99_03

    - THIS IS DEVELOPER RELEASE!
    - nick group feature was moved core to Plugin::HTMLFilter::NickGroup
    - App::Mobirc::HTTPD::* => App::Mobirc::Web::*
    - HTMLFilter::DoCoMoCSS is now works well

0.99_02

    - THIS IS DEVELOPER RELEASE!
    - Plugin::HTMLFilter::DoCoMoCSS was broken :(
    - use MooseX::Plaggerize!
    -- easy to write plugins.
    -- maintenable plugins!
    - Plugin::HTMLFilter::StickyTime was renamed to Plugin::StickyTime
    -- and load automatically
    - added many test cases.
    - Plugin::Component::XMPP is no longer supported
    -- you can use XIRCD!

0.99_01

    - THIS IS DEVELOPER RELEASE!
    - some feature is broken(e.g. DoCoMoCSS)
    - this release makes more maintenable code, not feature enhancement release.
    -- more mooooooooooooooooooooose
    -- use HTTP::Engine
    -- use HTTPx::Dispatcher
    -- use Template::Declare

0.11

    (no feature changes)
    - no singleton

0.10

    - use Mooooooooooooooooose
    - added Authorizer::DoCoMoGUID by walf443++
    - fixed dependencies.

0.06

    - fixed some tests.
    - fixed dependencies

0.05

    - (SECURITY FIX)fixed xss(http://coderepos.org/share/changeset/6204)
    - use more pictograms in mobile template.
    - give accesskey="9" to clear_all_unread.
    - added App/Mobirc/Plugin/DocRoot.pm
    - well formed html

0.04

    - fixed docroot related bug.fixed by hirose31++
    - added documents.
    - added tests

0.03

    initial release to CPAN.



Makefile.PL
use inc::Module::Install;
name 'mobirc';
all_from 'lib/App/Mobirc.pm';
author 'Tokuhiro Matsuno';

# ~/perl5/perlbrew/perls/perl-5.8.1/bin/perl -I extlib/lib/perl5/ tools/cpanm -nv -L extlib --installdeps .

license  'perl';

# AnyEvent
requires 'AnyEvent' => 5.271;
requires 'AnyEvent::IRC' => 0.95;

# Plack
requires 'Plack'                           => 0.9938;
requires 'Twiggy'                          => 0.1005;
requires 'Plack::Request'                  => 0.09;
requires 'Plack::Middleware::ReverseProxy' => 0.14;
requires 'Tatsumaki'                       => 0.1010;

# Mouse.
requires 'Mouse'         => 0.60;
requires 'MouseX::Types' => 0.05;
requires 'Any::Moose'    => 0.13;

# session
requires 'HTTP::Session' => '0.44';

# mobile stuff
requires 'HTTP::MobileAttribute' => '0.21';

# template engine
requires 'Text::MicroTemplate' => '0.13';

# utilities
requires 'parent'            => 0.223;
requires 'Config::Tiny'      => 2.12;
requires 'String::CamelCase' => '0.01';
requires 'Exporter' => '5.62';    # recent version supports "use Exporter 'import'" style
requires 'HTML::Entities'        => '1.35';
requires 'HTML::StickyQuery'     => '0.12';
requires 'Router::Simple'        => '0.05';
requires 'JSON'                  => '2.09';
requires 'List::MoreUtils'       => '0.22';
requires 'MIME::Base64::URLSafe' => '0.01';
requires 'Module::Find'          => '0.06';
requires 'Params::Validate'      => '0.91';
requires 'Path::Class'           => '0.19';
requires 'UNIVERSAL::require'    => '0.11';
requires 'URI'                   => '1.36';
requires 'URI::Find'             => '20100505';
requires 'YAML'                  => '0.68';
requires 'String::IRC'           => '0.04';
requires 'CSS::Tiny'             => '1.15';
requires 'Data::OptList'         => '0.105';
requires 'Params::Util'; # for Data::OptList
requires 'Data::Recursive::Encode';
requires 'Text::VisualWidth::PP' => 0.01;
requires 'JavaScript::Value::Escape' => 0;

features(
    'Irssi' => [
        -default => 0,
        recommends('POE::Session::Irssi', '0.4'),
        recommends('Glib', '0.4'),
        recommends('POE::Loop::Glib', '0.0034'),
    ],
);

tests 't/*.t t/*/*.t t/*/*/*.t t/*/*/*/*.t';
requires 'Test::More' => 0.94;
test_requires 'Test::More';
test_requires 'Text::Diff';
test_requires 'Test::Requires';
test_requires 'Test::Base::Less';
author_tests('xt');
install_script 'mobirc';
WriteAll;

